tests/unit/test_cv_device_tools.py

```diff
@@ -0,0 +1,133 @@
+from tests.data.device_tools_unit import device_data, current_container_info, cv_data, image_bundle
+from ansible_collections.arista.cvp.plugins.module_utils.device_tools import DeviceInventory, CvDeviceTools
+
+
+# list of paths to patch
+MOCK_LIST = [
+    'ansible_collections.arista.cvp.plugins.module_utils.device_tools.AnsibleModule',
+    'ansible_collections.arista.cvp.plugins.module_utils.device_tools.CvDeviceTools._CvDeviceTools__get_device',
+    'ansible_collections.arista.cvp.plugins.module_utils.device_tools.CvDeviceTools.get_container_current']
+
+@pytest.fixture
+def setup(apply_mock, mock_cvpClient):
+    """
+    setup - setup method to apply mocks and patches
+    """
+    mock_ansible_module, mock__get_device, mock_get_container_current = apply_mock(MOCK_LIST)
+
+    mock_ansible_module.fail_json.side_effect = mock_cvpClient.api.fail_json
+
+    cv_tools = CvDeviceTools(mock_cvpClient, mock_ansible_module, 'serialNumber')
+
+    return mock_ansible_module, mock__get_device, cv_tools, mock_get_container_current
+
+
+@pytest.mark.state_present
+class TestApplyBundle():
+    """
+    Contains unit tests for apply_bundle()
+    """
+    @pytest.mark.parametrize(
+        "expected",
+        [
+            (True),    #  success
+            (False),   # failure
+        ],
+    )
+    def test_apply_bundle(self, setup, expected):
+        """
+        Test when current_image_bundle and assigned_image_bundle are different
+
+        :param setup: fixture
+        :param expected: output expected from apply_bundle
+
+        """
+        user_topology = DeviceInventory(data=device_data)
+        _, mock__get_device, cv_tools, mock_get_container_current = setup
+        mock_get_container_current.return_value = current_container_info
+        mock__get_device.return_value = cv_data
+        if not expected:
+            # removing node_id for failure test-case
+            image_bundle['id'] = None
+
+        result = cv_tools.apply_bundle(user_inventory=user_topology)
+        assert result[0].success == expected
+        assert result[0].changed == expected
+
+        # resetting imageBundle id
+        image_bundle['id'] = 'imagebundle_1658329041200536707'
+
+    def test_apply_bundle_same_image(self, setup):
+        """
+        Test when current_image_bundle and assigned_image_bundle are same
+        :param setup: fixture
+
+        if both image bundles are same, nothing to do.
+        """
+        user_topology = DeviceInventory(data=device_data)
+        _, mock__get_device, cv_tools, mock_get_container_current = setup
+        mock_get_container_current.return_value = current_container_info
+
+        cv_data_same = deepcopy(cv_data)
+        cv_data_same['imageBundle']['bundleName'] = device_data[0]['imageBundle']
+        mock__get_device.return_value = cv_data_same
+
+        result = cv_tools.apply_bundle(user_inventory=user_topology)
+        assert result[0].success == False
+        assert result[0].changed == False
```

Comment:
according to pylint suggestion
```suggestion
        assert result[0].success is False
        assert result[0].changed is False
```